cmd/dotmesh-server/s3_handlers.go

```diff
@@ -205,6 +208,71 @@ func (s *S3Handler) mountFilesystemSnapshot(filesystemId string, snapshotId stri
 	return e
 }
 
+func (s *S3Handler) headFile(resp http.ResponseWriter, req *http.Request, filesystemId, snapshotId, filename string) {
+	user := auth.GetUserFromCtx(req.Context())
+	fsm, err := s.state.InitFilesystemMachine(filesystemId)
+	if err != nil {
+		http.Error(resp, "failed to initialize filesystem", http.StatusInternalServerError)
+		return
+	}
+
+	state := fsm.GetCurrentState()
+	if state != "active" {
+		http.Error(resp, fmt.Sprintf("please try again later, state was %s", state), http.StatusServiceUnavailable)
+		return
+	}
+
+	// we must first mount the given snapshot before we try to read a file within it
+	// if snapshotId is not given then the latest snapshot id will be used
+	e := s.mountFilesystemSnapshot(filesystemId, snapshotId)
+	// the snapshot has been mounted - pass the SnapshotMountPath via the
+	// OutputFile to the fileOutputIO channel to get handled
+	if e.Name == "mounted" {
```

Comment:
I know this is likely taken from the other func which doesn't do early return, but since go's style is to early return maybe change this to
```
if e.Name != "mounted" {
// chunk of code from the else block below
}
// chunk of code from inside the if statement
```